### PR TITLE
Normalizing sonata_type_native_collection - styles and options

### PR DESCRIPTION
**Pull request by joshlopes**
Q   |   A
------|-------
Bug fix? |	no
New feature? |	no
BC breaks? |	no
Deprecations? |	no
Tests pass? | yes
Fixed tickets | no
License | no
Doc PR | no

Just adding correct styles and the options Sonata has on most of form types to the native_collection

EDIT by @oskarstark to embed the screenshot:

*BEFORE:*
bildschirmfoto 2016-02-12 um 17 04 34

*AFTER:*
bildschirmfoto 2016-02-12 um 17 04 53

### CODE REVIEW

**Comment by joshlopes** [comment]
can this be added ? so i can update my version without breaking stuff around :)

**Comment by Soullivaneuh** [comment]
Can you explain the point of this PR?

Please add some concrete case and example.

And also, if it's a new feature, please write some notes on documentation.

**Comment by joshlopes** [comment]
Its not a new feature -- or at least not in full. What it does is add the possibility to translate the buttons on native collections and make them to appear in the styles that Sonata uses for other cases.

**Comment by Soullivaneuh** [comment]
Ok I see. Will start a review.

File: Form/Type/CollectionType.php [hunk 1]
```diff
@@ -31,6 +34,30 @@ public function getParent()
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
```
Comment (Soullivaneuh) [inline Form/Type/CollectionType.php]: Please move `buildView` and `setDefaultOptions` on top.

`getParent` should be placed after them.

File: Form/Type/CollectionType.php [hunk 2]
```diff
@@ -31,6 +34,30 @@ public function getParent()
...
+    {
+        $view->vars['btn_add'] = $options['btn_add'];
+        $view->vars['btn_delete'] = $options['btn_delete'];
+        $view->vars['btn_catalogue'] = $options['btn_catalogue'];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
```
Comment (Soullivaneuh) [inline Form/Type/CollectionType.php]: Please prevent it from deprecation error.

Do something like this class: https://github.com/sonata-project/SonataAdminBundle/blob/master/Form/Type/ModelType.php#L65-L78

File: Form/Type/CollectionType.php [hunk 3]
```diff
@@ -31,6 +34,30 @@ public function getParent()
...
+    {
+        parent::setDefaultOptions($resolver);
```
Comment (Soullivaneuh) [inline Form/Type/CollectionType.php]: This is not needed.

File: Form/Type/CollectionType.php [hunk 4]
```diff
@@ -31,6 +34,30 @@ public function getParent()
...
+        $resolver->setDefaults(array(
+            'auto_initialize' => false,
```
Comment (Soullivaneuh) [inline Form/Type/CollectionType.php]: Why this option?

**Comment by Soullivaneuh** [comment]
Can you please provide a use case in order to test it with admin class and necessary code?

**Comment by joshlopes** [comment]
I am using SonataDoctrineMongo but that doesnt matter for this purpose:

````
#ProductBundle\Document\Product

/** ODM\Collection() **/
private $gallery;

public function __construct()
{
    $this->gallery = new ArrayCollection();
}

/**
     * @return ArrayCollection
     */
    public function getGallery()
    {
        return $this->gallery;
    }

    public function addGallery($img)
    {
        $this->gallery->add($img);
    }

   public function setGallery(ArrayCollection $gallery)
   {
        $this->gallery = $gallery;
   }

````

````
#ProductBundle\Admin\ProductAdmin

protected function configureFormFields(FormMapper $formMapper)
{
    $formMapper->add('gallery', 'collection', [
                    'allow_add' => true,
                    'btn_add' => 'Add Field',
                    'allow_delete' => true,
                    'required' => false
                ])
}
````

This should be enough to test this

**Comment by joshlopes** [comment]
Any news on this @Soullivaneuh ?

**Comment by OskarStark** [comment]
@joshlopes can you provide some before/after screenshots?

**Comment by joshlopes** [comment]
@OskarStark 

before:  http://grab.by/KsMO
after: http://grab.by/KsNS

I have also merged with the last official master (there was no conflicts so)

**Comment by OskarStark** [comment]
is this then already used in the datagrid filters?

**Comment by joshlopes** [comment]
@OskarStark i am not following, this has nothing to do with filters nor logic, is pure cosmetic. Allowing the user to add custom text to a button and rendering it properly on the view. again for "collection" only aka "sonata_native_collection"

**Comment by OskarStark** [comment]
Ok thank you

Am Mittwoch, 16. September 2015 schrieb Jose :

File: Tests/Form/Type/CollectionTypeTest.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+{
+
+    public function testGetDefaultOptions()
+    {
+        $type = new CollectionType();
+
+        $optionResolver = new OptionsResolver();
+
+        $type->setDefaultOptions($optionResolver);
+        $options = $optionResolver->resolve(
```
Comment (core23) [inline Tests/Form/Type/CollectionTypeTest.php]: Should be ``$options = $optionResolver->resolve(array( ``

**Comment by core23** [comment]
Please squash & rebase @joshlopes

**Comment by OskarStark** [comment]
i added the screenshots to the initial PR

**Comment by core23** [comment]
Are you still working on this? Please squash and rebase or close this PR @joshlopes

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**